pkg/server/registration.go

```diff
@@ -152,7 +152,8 @@ func (s *registrationServer) CreateJoinToken(
 
 	_, err := ds.RegisterToken(req)
 	if err != nil {
-		return request, err
+		s.l.Error(err)
+		return nil, errors.New("Error trying to register your token")
```

Comment:
Should we include the original error in this message?

Reply:
Is it safe to expose the full error? 
Personally, I prefer to return a custom error while logging the full error for debugging.